rpc/src/v1/impls/eth_pubsub.rs

```diff
@@ -306,7 +347,8 @@ impl<C: Send + Sync + 'static> EthPubSub for EthPubSubClient<C> {
 		let res = self.heads_subscribers.write().remove(&id).is_some();
 		let res2 = self.logs_subscribers.write().remove(&id).is_some();
 		let res3 = self.transactions_subscribers.write().remove(&id).is_some();
+		let res4 = self.sync_subscribers.write().remove(&id).is_some();
 
-		Ok(res || res2 || res3)
+		Ok(res || res2 || res3 || res4)
```

Comment:
Unrelated to this PR:

hmm, are we trying to remove `the subscription` from all possible subscriptions because we don't which one it is?
Thus, if we have a some logic bug i.e, adding a subscription to the wrong one this will still return true?